abap-code-review/src/tools.adoc

```diff
@@ -0,0 +1,122 @@
+== Tools
+Various tools can be used to orchestrate the code review process, this section introduces the tooling required for conducting code reviews in the context of ABAP development.
+
+=== Git
+link:https://en.wikipedia.org/wiki/Git[Git] is a popular link:https://en.wikipedia.org/wiki/Distributed_version_control[distributed version control system], which helps developers work on the same codebase. Unlike ABAP, Git works with files instead of objects in a database, which allows editing any file with
+any editor, and keeping multiple copies (branches) of files.
+
+At a technical level Git is a protocol, and is typically used in connection with cloud-based hosting platforms like:
+
+* link:https://github.com[GitHub]
+* link:https://gitlab.com[GitLab]
+* link:https://bitbucket.com[Bitbucket]
+* link:https://azure.microsoft.com/en-us/services/devops/repos/[Azure Repos]
+
+For an introduction to Git see:
+
+* link:https://git-scm.com/doc[Git Documentation]
+* link:https://guides.github.com/introduction/git-handbook/[Git Handbook]
+
+=== Code Review Platforms
+Git helps keeping track of changes to files, and the Git hosting platforms add extra functionality on top of that. As explained in the <<_introduction>>, the most important feature is *proposed commits*, which is the basis for code reviews.
```

Comment:
Yes, commits are the most important features and thus one should learn how to make good commits:
- https://chris.beams.io/posts/git-commit/

The most valuable asset of git repositories is the "history" and without good commit messages, the history is useless.

Reply:
yea, suggest a contribution regarding commit messages for the "Best Practices" chapter after the PR has been merged

Reply:
...besides, individual commits become irrelevant if you squash them upon merging PRs. See my previous commit above.